R/copy-model-from.R

```diff
@@ -222,17 +222,16 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,
   if (.update_model_file) {
 
     # read parent control stream
-    mod_str <- .parent_model_path %>% read_file()
+    mod_str <- nmrec::read_ctl(.parent_model_path)
 
     # replace the $PROBLEM line(s)
     new_mod_id <- get_model_id(.new_model_path)
-    mod_str <- str_replace(mod_str,
-                           "\\$PROB(.|\n)*?\\$",
-                           as.character(glue("$PROBLEM From bbr: see {new_mod_id}.yaml for details\n\n$")))
-    glue("")
+    prob_rec <- nmrec::select_records(mod_str, "prob")[[1]]
```

Comment:
Control streams must have a problem record, and nmrec requires there only be a single record. So, if there weren't one record, the `nmrec::read_ctl` call a few lines up would fail.

Still, it'd be nice if this was a little more defensive and abort if things looked unexpected. For example, something like this (untested):

```diff
diff --git a/R/copy-model-from.R b/R/copy-model-from.R
index 584ca3b3..b63100a5 100644
--- a/R/copy-model-from.R
+++ b/R/copy-model-from.R
@@ -226,8 +226,11 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,

     # replace the $PROBLEM line(s)
     new_mod_id <- get_model_id(.new_model_path)
-    prob_rec <- nmrec::select_records(mod_str, "prob")[[1]]
-    prob_str <- nmrec::get_record_option(prob_rec, "text")
+    prob_recs <- nmrec::select_records(mod_str, "prob")
+    if (length(prob_recs) != 1) {
+      dev_error("select_records should always return a single $PROBLEM record")
+    }
+    prob_str <- nmrec::get_record_option(prob_recs[[1]], "text")
     prob_str$value <- as.character(glue("From bbr: see {new_mod_id}.yaml for details"))

     # write parent control stream
```